packages/micropip/src/micropip/externals/pip/_internal/network/lazy_wheel.py

```diff
@@ -0,0 +1,257 @@
+"""Adapted from
+https://github.com/pypa/pip/blob/main/src/pip/_internal/network/lazy_wheel.py"""
```

Comment:
Could you indicate the version used (or commit hash)? Maybe in `packages/micropip/src/micropip/externals/pip/__init__.py` and then this header is no longer necessary. The link to master is going to be non accurate in a few months/years anyway.